netmap/netmap_contract.go

```diff
@@ -601,6 +605,32 @@ func ListConfig() []record {
 	return config
 }
 
+// SubscribeForNewEpoch registers passed contract as a NewEpoch event
+// subscriber. Such a contract must have a `NewEpoch` method with a single
+// numeric argument. Transactions that call SubscribeForNewEpoch must be
+// witnessed by the Alphabet.
+// Produces `NewEpochHookRegistration` notification event with a just
+// registered recipient in a success case.
+func SubscribeForNewEpoch(contract interop.Hash160) {
```

Comment:
Just a question: don't we need unsubscription functionality?

Reply:
currently no. contracts should always be aware. mb for hotfixes only. in the future, i think, it is possible to extend (if needed) with a single update (a version that needs unsubscription also adds the unsubscribe method)